src/sql/inspect.py

```diff
@@ -73,6 +75,170 @@ def __init__(self, name, schema, conn=None) -> None:
         self._table_txt = self._table.get_string()
 
 
+@modify_exceptions
+class TableDescription(DatabaseInspection):
+    """
+    Generates descriptive statistics.
+
+    Descriptive statistics are:
+
+    Count - Number of all non empty values
+
+    Mean - Mean of the values
+
+    Max - Maximum of the values in the object.
+
+    Min - Minimum of the values in the object.
+
+    STD - Standard deviation of the observations
+
+    25h, 50h and 75h percentiles
+
+    Unique - Number of unique values
+
+    Top - The most frequent value
+
+    Freq - Frequency of the top value
+
+    """
+
+    def __init__(self, table_name, schema=None, config=None, user_ns=None) -> None:
+
+        if schema:
+            table_name = f"{schema}.{table_name}"
+
+        result_table_columns = sql.run.run(
+            Connection.current, f"SELECT * FROM {table_name} WHERE 1=0", config, user_ns
+        )
+
+        columns = result_table_columns.keys
+
+        table_stats = dict({})
+
+        for column in columns:
+            table_stats[column] = dict()
+            result_col_unique_values = sql.run.run(
+                Connection.current,
+                f"SELECT COUNT(DISTINCT {column}) as unique_count FROM {table_name}",
+                config,
+                user_ns,
+            )
+
+            result_col_freq_values = sql.run.run(
+                Connection.current,
+                f"""SELECT {column}, COUNT({column}) as frequency FROM {table_name}
+                GROUP BY {column} ORDER BY Count({column}) Desc""",
+                config,
+                user_ns,
+            )
+
+            result_non_empty_values = sql.run.run(
+                Connection.current,
+                f"""SELECT {column} FROM {table_name} WHERE {column}
+                IS NOT NULL AND TRIM({column}) <> ''
+                ORDER BY {column} ASC
+                """,
+                config,
+                user_ns,
+            )
+
+            col_values = result_non_empty_values.dict()[column]
```

Comment:
looks like this is fetching all the data right? this is unfeasible as users might profile tables with millions of rows. we should use `AVG` for the mean, and `COUNT(*)` for counting; the idea is that we should only fetch the aggregated data to the client, everything else should happen in the database